src/crystal/compiler_rt/mul.cr

```diff
@@ -0,0 +1,41 @@
+# :nodoc:
```

Comment:
Can you please add references to the original implementations in this file, too?